### PR TITLE
Merge pkgci_test_ jobs using matrices.

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/iree-org/iree/issues/17136. Similar to https://github.com/iree-org/iree/pull/17509, this folds the five `regression_test_` jobs in pkgci.yml into two matrices, `test_onnx` and `test_models`, with most steps shared.

The "model" tests are still verbose to configure here as we have multiple variants on some backends. Depending on how much we want to unify or diverge we could continue to refactor further.

ci-exactly: build_packages,regression_test

### CODE REVIEW

File: .github/workflows/pkgci_regression_test.yml [hunk 1]
```diff
@@ -44,63 +80,76 @@ jobs:
...
+            sdxl-config-file: sdxl_scheduled_unet_cpu_llvm_task.json
+            runs-on: nodai-amdgpu-w7900-x86-64
+
+          # AMD GPU
+          - name: amdgpu_rocm_gfx90a
+            models-config-file: pytorch_models_gpu_rocm_gfx90a.json
+            models-extra-flags-config-file: pytorch_models_gpu_rocm_gfx90a_additional_flags.json
+            sdxl-config-file: sdxl_scheduled_unet_gpu_rocm_gfx90a.json
+            runs-on: nodai-amdgpu-mi250-x86-64
+          - name: amdgpu_vulkan
```
Comment (saienduri) [inline .github/workflows/pkgci_regression_test.yml:150]: add new line and #NVIDIA GPU label

**Review by saienduri — APPROVED** [review]
LGTM, thanks

File: .github/workflows/pkgci_regression_test.yml [hunk 2]
```diff
@@ -161,25 +221,25 @@ jobs:
...
+            --log-cli-level=info \
             --timeout=1200 \
-            --retries 2 \
-            --retry-delay 5 \
             --durations=0 \
-            --config-files=build_tools/pkgci/external_test_suite/sdxl_scheduled_unet_gpu_rocm_gfx90a.json
+            --config-files=${SDXL_CONFIG_FILE_PATH}
+
       - name: "Running SDXL rocm pipeline benchmark"
+        if: contains(matrix.name, 'rocm')
```
Comment (saienduri) [inline .github/workflows/pkgci_regression_test.yml:260]: want to add the same success check here or I can do it in my PR too

Comment (ScottTodd) [reply .github/workflows/pkgci_regression_test.yml:260]: I'm not quite sure how much I want to include other conditions yet, so leaving that off.

File: .github/workflows/pkgci_regression_test.yml (hunk 1 above)
Comment (ScottTodd) [reply .github/workflows/pkgci_regression_test.yml:150]: 👍 added 
```
          # NVIDIA GPU
          # None at the moment. Could maybe use the persistent a100 runners:
          #   - self-hosted # must come first
          #   - runner-group=${{ needs.setup.outputs.runner-group }}
          #   - environment=${{ needs.setup.outputs.runner-env }}
          #   - a100
          #   - os-family=Linux
          # (note: would need to plumb the presubmit/postsubmit runner-group through to here too)
```

File: .github/workflows/pkgci_regression_test.yml [hunk 3]
```diff
@@ -44,63 +80,85 @@ jobs:
...
           python -m pip install -r SHARK-TestSuite/iree_tests/requirements.txt
+
       - name: Run external tests - ONNX test suite
         run: |
           source ${VENV_DIR}/bin/activate
-          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/rocm/lib:/opt/rocm/hip/lib
           pytest SHARK-TestSuite/iree_tests/onnx/ \
-              -rpfE --timeout=30 --durations=20 \
-              --config-files=build_tools/pkgci/external_test_suite/onnx_gpu_rocm_rdna3.json \
+              -n 4 -rpfE --timeout=30 --durations=20 \
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test.yml]: Oooh wait, we were running without `-n` at all on ROCm, right right... I can make that configurable via the matrix too - still want to run with 4 or more in parallel on at least CPU.

Comment (ScottTodd) [reply .github/workflows/pkgci_regression_test.yml]: Pushed a commit for that. Now we can at least see in one location how the jobs differ.

**Pull request merged**